ipfs-9-to-10/migration/migration.go

```diff
@@ -0,0 +1,146 @@
+package mg9
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	migrate "github.com/ipfs/fs-repo-migrations/go-migrate"
+	lock "github.com/ipfs/fs-repo-migrations/ipfs-1-to-2/repolock"
+	mfsr "github.com/ipfs/fs-repo-migrations/mfsr"
+	log "github.com/ipfs/fs-repo-migrations/stump"
+)
+
+type Migration struct{}
+
+func (m Migration) Versions() string {
+	return "9-to-10"
+}
+
+func (m Migration) Reversible() bool {
+	return true
+}
+
+func (m Migration) Apply(opts migrate.Options) error {
+	log.Verbose = opts.Verbose
+	log.Log("applying %s repo migration", m.Versions())
+
+	log.VLog("locking repo at %q", opts.Path)
+	lk, err := lock.Lock2(opts.Path)
+	if err != nil {
+		return err
+	}
+	defer lk.Close()
+
+	repo := mfsr.RepoPath(opts.Path)
+
+	log.VLog("  - verifying version is '9'")
+	if err := repo.CheckVersion("9"); err != nil {
+		return err
+	}
+
+	basepath := filepath.Join(opts.Path, "config")
+	v9path := filepath.Join(opts.Path, "config-v9")
+	if err := os.Rename(basepath, v9path); err != nil {
```

Comment:
We should:

1. Create a temporary file using `atomicfile` (see how go-ipfs-config does this).
2. Write the new config into that temporary file.
3. Atomically rename it.

Reply:
That way we're never in an inconsistent state.